tutorials/marble/version-control.md

```diff
@@ -0,0 +1,393 @@
+git config --global user.email "email@email.com"
+```
+
+### Staging a File
+
+Make sure you are in the same directory as the file you want to stage.
+
+```
+git add <filename>
+```
+
+### View Staged Files
+
+If you have multiple files staged and ready to commit you can see the list by using `git status`.
+
+```
+git status
+```
+
+
+### Making a Commit
+
+Add a brief descriptive message about your commit.
+
+```
+git commit -m "your brief commit message goes here"
+```
+
+### Pushing a Commit
+
+After your commit is complete, push it to the remote repository
+
+```
+git push
+```
+
+### Stashing Files
+
+Make sure you are in the same directory as the file you want to stash.  
+
+Running the `stash` command will stash all files that have changes made to them.
+
+```
+git stash
+```
+
+To see the files you stashed you can list them: 
+
+```
+git stash list
+```
+
+To resume working on your stashed files you will need to get them out of the stash.  Run the following command and you will see them under the `Changed` section.
+
+```
+git stash pop
+```
+
+
+### Creating a Branch
+
+```
+git branch <new-branch-name>
+```
+
+### Switching Branches
+
+```{note}
+You can only switch branches when you have no files staged or changed.
+``` 
+
+Switch branches by checking out the branch.  Replace *<branch-name>* with the branch you want to switch to.
+
+```
+git checkout <branch-name>
+```
+
+```{note}
+You can only switch to another local branch.  Branches with the `origin` prefix are remote branches and cannot be switched into.
+``` 
+
+## Creating a Pull Request
+A Pull Request is done when you want to merge your changes into the **master**/**main** branch.  It also gives you a chance to let others know of the changes you made and ask them to review your changes.
+
+A Pull Request is done from the GitHub site.  After logging into [GitHub](https://www.github.com), you may see a notification about a branch making recent pushes if you pushed recently.
+
+![GitHub PR Notice](images/version-control/github-pr-notice.png)
+
+Click the `Compare & pull request` button.  
+
+If you don't see a notification, click the `Pull Request` menu button and then the `New Pull Request` button on the following page.
+
+![GitHub PR Menu](images/version-control/github-pr-menu.png)  ![GitHub New PR](images/version-control/github-new-pr.png)
+
+You'll be taken to the `Open a pull request` page.
+
+Here, make sure the `base` is set to **master** and the `compare` is set to the branch you want to compare the **master** branch with.
+
+![GitHub Base Compare](images/version-control/github-base-compare.png)
+
+Fill out the form and provide a title and description of the changes made.  
+
+If you have someone who can review your pull request click the cog icon in the right side of the screen under the `Reviewers` section and select their name.
+
+![GitHub Reviewers](images/version-control/github-reviewers.png)
+
+Then click the `Create Pull Request` button.
+
+## Merging Your Branch
```

Comment:
Move this into a separate section about GitHub specifically

Also maybe add a section on merging branches on the command line (in the relevant section above)

Reply:
Moved the Create Pull Request section under the GitHub section.